packages/SwingSet/docs/virtual-objects.md

```diff
@@ -0,0 +1,111 @@
+# Vat Secondary Storage
+
+The kernel currently provides two secondary storage mechanisms for the use of (application) code running in a vat:
+
+- Virtual objects
+- Persistent weak stores
+
+Each of these is accessed via a global made available to vat code.
+
+## Virtual Objects
+
+A virtual object is an object with a durable identity whose state is state is automatically and transparently backed up in secondary storage.  This means that when a given virtual object is not in active use, it need not occupy any memory within the executing vat.  Thus a vat can manage an arbitrarilly large number of such objects without concern about running out of memory.
+
+A virtual object has a "kind", which defines what sort of behavior and state it will possess.  You can think of a kind somewhat like a data type, but without direct linguistic support.
```

Comment:
Yes, I was a little hesitant about my phrasing also.  You're right that it's not a data type, which is why described it as _like_ a data type, hoping to draw an analogy.  It is a type of type, but it's not abstract.  One might almost want to use the world "class" if it weren't already hopelessly overloaded.  "Kind" means "the things with _this_ implementation", but I don't care for the description in terms of the objects-as-closure pattern, because I don't want to have to describe the objects-as-closure pattern nor assume people know what that means. I think this paragraph still needs work.

Reply:
Try this on for size:

A virtual object has a "kind", which defines what sort of behavior and state it will possess.  A kind is not exactly a data type, since it comes with a concrete implementation, but it indicates a family of objects that share a set of common behaviors and a common state template.